.github/workflows/_buildpacks-release.yml

```diff
@@ -0,0 +1,245 @@
+        uses: buildpacks/github-actions/setup-pack@v5.3.0
+
+      - name: Save CNB File
+        run: pack buildpack package ${{ matrix.buildpack_artifact_prefix }}.cnb --config ${{ matrix.buildpack_output_dir }}/package.toml --format file -v
+
+      - name: Get token for GitHub application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ inputs.app_id }}
+          private_key: ${{ secrets.app_private_key }}
+
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v0.1.15
+        with:
+          token: ${{ steps.generate-token.outputs.app_token }}
+          tag_name: v${{ matrix.buildpack_version }}
+          body: ${{ needs.compile.outputs.changelog }}
+          files: ${{ matrix.buildpack_artifact_prefix }}.cnb
+          fail_on_unmatched_files: true
+
+  publish-cnb:
+    name: Publish → CNB Registry - ${{ matrix.buildpack_id }}
+    needs: [compile, publish-docker]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJSON(needs.compile.outputs.buildpacks) }}
+    steps:
+      - name: Install crane
+        uses: buildpacks/github-actions/setup-tools@v5.3.0
+
+      - name: Check if version is already in the registry
+        id: check
+        run: |
+          export URL="https://registry.buildpacks.io/api/v1/buildpacks/${{ matrix.buildpack_id }}/${{ matrix.buildpack_version }}"
+          export EXISTS=$(if [ "$( curl -s -o /dev/null -I -w "%{http_code}" "${URL}")" = "200" ]; then echo 'true'; else echo 'false'; fi)
+          echo "published_to_cnb_registry=${EXISTS}" >> $GITHUB_OUTPUT
+
+      - name: Calculate the buildpack image digest
+        id: digest
+        run: echo "value=$(crane digest ${{ matrix.docker_repository }}:${{ matrix.buildpack_version }})" >> "$GITHUB_OUTPUT"
+
+      - name: Register the new version with the CNB Buildpack Registry
+        if: inputs.dry_run == false && steps.check.outputs.published_to_cnb_registry == 'false'
+        uses: docker://ghcr.io/buildpacks/actions/registry/request-add-entry:5.2.0
+        with:
+          token: ${{ secrets.cnb_registry_token }}
+          id: ${{ matrix.buildpack_id }}
+          version: ${{ matrix.buildpack_version }}
+          address: ${{ matrix.docker_repository }}@${{ steps.digest.outputs.value }}
+
+  update-builder:
+    name: Update Builder
+    needs: [compile, publish-docker, publish-cnb, publish-github]
+    runs-on: ${{ inputs.ip_allowlisted_image }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: ./buildpacks
+
+      - name: Checkout builder repository
+        uses: actions/checkout@v3
+        with:
+          repository: heroku/builder
+          path: ./builder
+
+      - name: Install crane
+        uses: buildpacks/github-actions/setup-tools@v5.3.0
+
+      - name: Update Builder
+        if: inputs.dry_run == false
```

Comment:
Should this `if: inputs.dry_run == false` conditional be moved to the "create pull request" step?

Whilst the "create pull request" step should skip itself if there are no changed files (so the current implementation won't write changes), moving the conditional would:
- make the intent here a bit easier to follow
- mean the update-builder command gets run/tested in dry run mode (whereas at the moment it doesn't run, so isn't tested)